### ISSUE TITLE
UTF-8 characters are broken in fsharpi

### ISSUE THREAD

**Issue by haf** [Microsoft/visualfsharp#2581]
#### Repro steps

Provide the steps required to reproduce the problem

TERM_PROGRAM_VERSION=3.0.4
TERM_PROGRAM=iTerm.app

```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
export LC_CTYPE=UTF-8
```
1. Start `fsharpi` on OS X
2. Type `ä` with a Swedish keyboard.
3. 
   <img width="138" alt="screen shot 2016-07-27 at 13 38 48" src="https://cloud.githubusercontent.com/assets/193115/17174147/a3073116-53ff-11e6-8f72-6ae70a8ad307.png">
#### Expected behavior

`ä` is displayed.
#### Actual behavior

`�` is displayed.
#### Known workarounds

No known.
#### Related information

Provide any related information 
- Operating system: OS X
- Branch:

```
➜  ~ fsharpi --version


error FS0243: Unrecognized option: '--version'
```
- .NET Runtime, CoreCLR or Mono Version

```
➜  ~ mono --version
Mono JIT compiler version 4.4.1 (mono-4.4.0-branch-c7sr0/4747417 Mon Jun 20 15:43:48 EDT 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
    TLS:           normal
    SIGSEGV:       altstack
    Notification:  kqueue
    Architecture:  x86
    Disabled:      none
    Misc:          softdebug
    LLVM:          yes(3.6.0svn-mono-master/a173357)
    GC:            sgen
```
- Editing Tools (e.g. Visual Studio Version): none
- Links to F# RFCs or entries on http://fslang.uservoice.com: none
- Links to performance testing scripts: none
- Indications of severity: medium

Ref https://github.com/fsharp/fsharp/issues/654

Ref https://github.com/ionide/ionide-vscode-fsharp/issues/290

Ref http://forum.pdfsharp.net/viewtopic.php?f=2&t=3553&p=10681#p10681

Ref https://github.com/fsharp/fsharp/issues/594

Ref https://github.com/Microsoft/visualfsharp/pull/2582

**Comment by aliozgur** [comment]
Same here.
<img width="1439" alt="2017-04-07_1812" src="https://cloud.githubusercontent.com/assets/799194/24806588/fe02b910-1bbd-11e7-9619-813d28933f23.png">

**Comment by haf** [comment]
This is (possibly) fixed in this repo, but there's no way to test it on OS X (no instructions on how to replace the FSI or change the underlying DLL). Instructions or hints on how to do that would be welcome.

**Comment by dsyme** [comment]
The fix was verified by people on fsharp/fsharp#654
